### PR TITLE
CXF-7291: sharedPackageName for Swagger desc should ignore proxys

### PR DESCRIPTION
**Pull request by deki**

### CODE REVIEW

**Comment by sberyozkin** [comment]
Hi Dennis, is it related to the client proxy thing ?

**Comment by deki** [comment]
yes but in my view this should be fixed in any case

File: core/src/main/java/org/apache/cxf/common/util/PackageUtils.java [hunk 1]
```diff
@@ -56,7 +57,9 @@ public static String getSharedPackageName(List<Class<?>> classes) {
         List<List<String>> lParts = new ArrayList<List<String>>(classes.size());
         List<String> currentParts = new ArrayList<>();
         for (Class<?> cls : classes) {
-            lParts.add(StringUtils.getParts(getPackageName(cls), "\\."));
+            if (!Proxy.isProxyClass(cls)) {
```
Comment (reta) [inline core/src/main/java/org/apache/cxf/common/util/PackageUtils.java]: In this case, should we get the real class using `getProxyClass` and extract the package from it?

Comment (deki) [inline core/src/main/java/org/apache/cxf/common/util/PackageUtils.java]: Good idea, I first thought to use InvocationHandler stuff for this but in the method only the class itself is available. And afaik it's not possible to get the real class as there might be multiple interfaces...

**Pull request closed without merging**